exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -471,6 +428,66 @@ func (b *Bitfinex) wsHandleData(respRaw []byte) error {
 	return nil
 }
 
+func (b *Bitfinex) handleWSEvent(respRaw []byte) error {
+	event, err := jsonparser.GetUnsafeString(respRaw, "event")
+	if err != nil {
+		return fmt.Errorf("error parsing WS event name: %w from message: %s", err, respRaw)
+	}
+	switch event {
+	case wsEventSubscribed:
+		subID, err := jsonparser.GetUnsafeString(respRaw, "subId")
+		if err != nil {
+			return fmt.Errorf("error parsing WS subscribed event subId: %w from message: %s", err, respRaw)
+		}
+		if !b.Websocket.Match.IncomingWithData("subscribe:"+subID, respRaw) {
+			return fmt.Errorf("%v channel subscribe listener not found", subID)
+		}
+	case wsEventUnsubscribed:
+		chanID, err := jsonparser.GetInt(respRaw, "chanId")
+		if err != nil {
+			return fmt.Errorf("error parsing WS unsubscribed event chanId: %w from message: %s", err, respRaw)
+		}
+		if !b.Websocket.Match.IncomingWithData("unsubscribe:"+strconv.Itoa(int(chanID)), respRaw) {
+			return fmt.Errorf("%v channel unsubscribe listener not found", chanID)
+		}
+	case wsEventError:
+		if subID, err := jsonparser.GetUnsafeString(respRaw, "subId"); err == nil {
+			if !b.Websocket.Match.IncomingWithData("subscribe:"+subID, respRaw) {
+				return fmt.Errorf("%v channel subscribe listener not found", subID)
+			}
+		} else if chanID, err := jsonparser.GetInt(respRaw, "chanId"); err == nil {
+			if !b.Websocket.Match.IncomingWithData("unsubscribe:"+strconv.Itoa(int(chanID)), respRaw) {
+				return fmt.Errorf("%v channel unsubscribe listener not found", chanID)
+			}
+		} else {
+			return fmt.Errorf("unknown channel error; Message: %s", respRaw)
+		}
+	case wsEventAuth:
+		status, err := jsonparser.GetUnsafeString(respRaw, "status")
+		if err != nil {
+			return fmt.Errorf("error parsing WS auth event status: %w from message: %s", err, respRaw)
+		}
+		if status == "OK" {
+			var glob map[string]interface{}
+			if err := json.Unmarshal(respRaw, &glob); err != nil {
+				return fmt.Errorf("unable to Unmarshal auth resp; Error: %w Msg: %v", err, respRaw)
+			}
+			// TODO - Send a better value down the channel
+			b.Websocket.DataHandler <- glob
+		} else {
```

Comment:
This else fixes the broken check for fail vs FAIL.
For now, I see no reason all statuses other than OK shouldn't be an error.

Reply:
Agreed